src/backend/XmlHttpRequest.ts

```diff
@@ -390,16 +415,6 @@ export default class XmlHttpRequest extends BaseFileSystem implements FileSystem
   }
 
   /**
-   * Asynchronously download the given file.
-   */
-  private _requestFileAsync(p: string, type: 'buffer', cb: BFSCallback<Buffer>): void;
-  private _requestFileAsync(p: string, type: 'json', cb: BFSCallback<any>): void;
-  private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void;
-  private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void {
-    asyncDownloadFile(this.getXhrPath(p), type, cb);
-  }
-
-  /**
    * Synchronously download the given file.
    */
   private _requestFileSync(p: string, type: 'buffer'): Buffer;
```

Comment:
I looked through the code, and it doesn't. I added methods like `supportsSynch` primarily for unit tests, and secondarily for users of the library that need to generically check what operations a backend supports. I relied on the `BaseFileSystem` throwing `ENOTSUP` errors for sync operations on async file systems, and never anticipated that a sync backend might be async in some cases. Plus, there are some cases where synchronous operations may or may not be supported depending on the path (e.g., a MountableFileSystem with sync and async backends).